lib/her/model/associations/belongs_to_association.rb

```diff
@@ -22,6 +22,16 @@ def #{name}
               cached_data = (instance_variable_defined?(cached_name) && instance_variable_get(cached_name))
               cached_data || instance_variable_set(cached_name, Her::Model::Associations::BelongsToAssociation.proxy(self, #{opts.inspect}))
             end
+
+            def #{name}=(value)
+              raise "AssociationTypeMismatch: #{opts[:class_name]} expected, got \#{value.class.name}" unless #{!!opts[:polymorphic]} || value.class.name == "#{opts[:class_name]}"
+              cached_name = :"@_her_association_#{name}"
+
+              if value.persisted?
+                send("#{opts[:foreign_key]}=", value.id)
+                instance_variable_set(cached_name, nil)
+              end
```

Comment:
I think we could change replace this if block with:
```
instance_variable_set(cached_name, value)
send("#{opts[:foreign_key]}=", value.id) if value.persisted?
```
What do you think? Obviously we'd need to try out some code again, but based on what we did in has_one_association.rb, I think this might be an ok change.